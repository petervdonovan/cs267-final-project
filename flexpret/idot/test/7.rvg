[mu []
[def (Areg) t0]
[def (A) [array-new Areg of length 32]]
[[file {Reset_Handler} {
  Reset_Handler:
    [id init]
    [single t5
    {
      [allocate 32 Areg]
      [for t1 t2 0 to [* 4 32] by 4
        {
          add t3 t1 Areg
          srli t4 t1 2
          sw t4 0(t3)
        }]
      [iterate {addr} of [A t2 t3]
        {
          lw t1 [id addr]
          [sim-print t1 t4]
        }]
      [iterate {t1} of [map mapload [A t2 t3]]
        {
          [sim-print t1 t4]
        }]
    }]
    [id die]}]]
]
