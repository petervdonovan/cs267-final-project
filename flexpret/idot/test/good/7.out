Core-0:          0
Core-0:          1
Core-0:          2
Core-0:          3
Core-0:          4
Core-0:          5
Core-0:          6
Core-0:          7
Core-0:          8
Core-0:          9
Core-0:         10
Core-0:         11
Core-0:         12
Core-0:         13
Core-0:         14
Core-0:         15
Core-0:         16
Core-0:         17
Core-0:         18
Core-0:         19
Core-0:         20
Core-0:         21
Core-0:         22
Core-0:         23
Core-0:         24
Core-0:         25
Core-0:         26
Core-0:         27
Core-0:         28
Core-0:         29
Core-0:         30
Core-0:         31
Core-0:          0
Core-0:          1
Core-0:          2
Core-0:          3
Core-0:          4
Core-0:          5
Core-0:          6
Core-0:          7
Core-0:          8
Core-0:          9
Core-0:         10
Core-0:         11
Core-0:         12
Core-0:         13
Core-0:         14
Core-0:         15
Core-0:         16
Core-0:         17
Core-0:         18
Core-0:         19
Core-0:         20
Core-0:         21
Core-0:         22
Core-0:         23
Core-0:         24
Core-0:         25
Core-0:         26
Core-0:         27
Core-0:         28
Core-0:         29
Core-0:         30
Core-0:         31
Core-0 is done
Core-1 is done
Core-2 is done
Core-3 is done
Core-4 is done
Core-5 is done
Core-6 is done
Core-7 is done
Core-8 is done
Core-9 is done
Core-10 is done
Core-11 is done
Core-12 is done
Core-13 is done
Core-14 is done
Core-15 is done
All cores are done terminating
Assertion failed: Program terminated sucessfully
